pkg/gofr/cron.go

```diff
@@ -62,22 +58,33 @@ type tick struct {
 }
 
 // NewCron initializes and returns new cron tab.
-func NewCron(cntnr *container.Container) *Crontab {
+func NewCron(cntnr *container.Container, options ...func(*Crontab)) *Crontab {
 	c := &Crontab{
 		ticker:    time.NewTicker(time.Second),
+		location:  time.Local,
 		container: cntnr,
 		jobs:      make([]*job, 0),
 	}
 
+	for _, option := range options {
+		option(c)
+	}
+
 	go func() {
 		for t := range c.ticker.C {
+			t = t.In(c.location)
 			c.runScheduled(t)
 		}
 	}()
 
 	return c
 }
 
+// WithTimezone is a functional option for NewCron to specify timezone against which the cron schedule will be evaluated.
+func WithTimezone(location *time.Location) func(*Crontab) {
+	return func(c *Crontab) { c.location = location }
+}
```

Comment:
Guys, what do you think about adding also something like this



```suggestion
// WithTimezone is a functional option for NewCron to specify timezone against which the cron schedule will be evaluated.
func WithTimezone(location *time.Location) func(*Crontab) {
	return func(c *Crontab) { c.location = location }
}

func WithTesting(t *testing.T) func(*Crontab) {
	return func(c *Crontab) { t.Cleanup(func() { c.ticker.Stop() })}
}
```

This could help us to avoid using the `defer c.ticker.Stop()` that is added everywhere in the tests

Reply:
Another solution could be to provide something like this

```suggestion
// WithTimezone is a functional option for NewCron to specify timezone against which the cron schedule will be evaluated.
func WithTimezone(location *time.Location) func(*Crontab) {
	return func(c *Crontab) { c.location = location }
}

func NewCronT(t *testing.T, cntnr *container.Container, options ...func(*Crontab)) *Crontab {
	c := NewCron(cntnr, options...)
	t.Cleanup(func() {
		c.ticker.Stop()
	}
	return c
}
```

This second option should better to me

Reply:
In the second option, does the `NewCronT` factory need to be exported, considering the test file resides in the same package ?

Reply:
I quoted and suggested where I initially started talking about it, but yes of course: there is no need to have it exported, also it should put in  the cron_test.go file.